website_compassion/controllers/my_account.py

```diff
@@ -269,6 +269,35 @@ def my_child(self, state="active", child_id=None, **kwargs):
             # No child is selected, we pick the first one by default
             return request.redirect(f"/my/children?child_id={children[0].id}")
 
+    @route("/my/donations/update", type="http", auth="user", website=True)
+    def my_donations_update(self, recurring_contract=None, res=None, **kw):
+        partner = request.env.user.partner_id
+        if not recurring_contract:
+            return request.redirect("/my/donations")
+        contract = (partner.contracts_fully_managed
+                    + partner.contracts_paid) \
+            .filtered(lambda a: a.state not in ["cancelled", "terminated"]).filtered(
+            lambda a: int(a.id) == int(recurring_contract))
```

Comment:
This is useless computations and filtering.
```suggestion
        contract = request.env["recurring.contract"].browse(int(recurring_contract))
```